src/decrypt.c

```diff
@@ -91,19 +97,27 @@ ece_decrypt_records(const uint8_t* key, const uint8_t* nonce, uint32_t rs,
   size_t recordStart = 0;
   size_t blockStart = 0;
   for (size_t counter = 0; recordStart < ciphertextLen; counter++) {
-    size_t recordEnd = recordStart + rs;
-    if (recordEnd > ciphertextLen) {
+    size_t recordEnd;
+    if (rs > ciphertextLen - recordStart) {
+      // This check is equivalent to `recordStart + rs > ciphertextLen`; it's
```

Comment:
I don't think `recordStart` can exceed `ciphertextLen`, the exit condition for the loop is `recordStart < ciphertextLen`. (I should also call this variable `ciphertextStart`, to make it clearer it's an offset into the ciphertext...and match the encryption code).